code/src/App.js

```diff
@@ -1,9 +1,140 @@
-import React from 'react'
+import React, { useState, useEffect, useCallback, useRef } from "react";
+
+import { API_URL, LIKES_URL } from "./reusable/urls";
+import { MessageList } from "./components/MessageList";
+import { MessageForm } from "./components/MessageForm";
 
 export const App = () => {
+  const [messageList, setMessageList] = useState([]);
+  const [newMessage, setNewMessage] = useState("");
+  const [userInput, setUserInput] = useState("");
+  const [keypressCount, setKeypressCount] = useState(0);
+  const [charRange, setCharRange] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const messageRef = useRef();
+
+
+  /*-------------fetches the list of all tweets--------------*/
+
+  const fetchMessageList = useCallback(() => {// callback to avoid re-rendering of dependencies
+    
+    fetch(API_URL)
+      .then(setLoading(true)) //to show loading symbol
```

Comment:
it would be nice to move it before the fetch and show it only if `messageList` is empty. It would prevent jumping of your messages if nothing new is posted 🤓